torchvision/datasets/folder.py

```diff
@@ -192,8 +210,14 @@ class ImageFolder(DatasetFolder):
     """
 
     def __init__(self, root, transform=None, target_transform=None,
-                 loader=default_loader):
-        super(ImageFolder, self).__init__(root, loader, IMG_EXTENSIONS,
-                                          transform=transform,
-                                          target_transform=target_transform)
+                 loader=default_loader, is_valid_file=None):
+        if is_valid_file is None:
```

Comment:
instead of duplicating this codepath, we can maybe follow a similar approach as in my first message, and only pass `is_valid_file`?

Reply:
@fmassa This can cause wrong (confusing) error message (stack) because error will come from is_valid_file even if user did not passed it . what do you think about this ?

Reply:
what kind of error messages are you thinking about?

Reply:
@fmassa  like if user does not pass a list of strings as extensions. that we did not handle so he will get error msg. anyways I think `extension=IMG_EXTENSIONS if is_valid_file is None else None` will be better one liner we need isn't it?

Reply:
But `IMG_EXTENSIONS` is not visible to the user, right? So he has no control on it anyway

Reply:
Ya I almost forgot that. still above mentioned way is correct right instead of creating a lambda here?

Reply:
This is fine as well, thanks

Reply:
@fmassa is did the changes you suggested please take a look.